src/components/Article.jsx

```diff
@@ -9,20 +9,25 @@ import formatDate from "../utils/formatDate";
 const Article = () => {
     const {article_id} = useParams();
     const {articleData, isLoading} = useArticle(article_id);
-    const [hasVoted, setHasVoted] = useState(false);
+    const [hasVoted, setHasVoted] = useState(Boolean(localStorage.getItem(`voted${article_id}`)));
+    const [error, setError] = useState(null);
 
     const upvotehandler = () => {
         const vote = {inc_votes: 1};
         articleData.votes += 1;
-        setHasVoted(true);
-        patchArticleById(article_id, vote);
+        localStorage.setItem(`voted${article_id}`, true);
```

Comment:
As a lot of this logic is repeated across these two handler functions, I'd consider making them a single one that takes an argument of 1 or -1. Also, don't forget to change upvotehandler to upvoteHandler for consistency